src/main/java/gov/nasa/podaac/swodlr/user/UserReference.java

```diff
@@ -10,6 +10,9 @@ public class UserReference implements Serializable {
   private static UserRepository userRepository;
 
   private final UUID id;
+  
+  // Attempt to cache here so future lookups don't hit database
```

Comment:
Another option would to enable the application-cache through Spring boot https://docs.spring.io/spring-boot/reference/io/caching.html

Just using the default built in concurrent hashmap would probably be equivalent to adding transient here but would open up the option to configure a proper app cache (like geode or redis) in the future

Reply:
Looks like there's a fair bit of work we'd need to do to get caching to work on UserReferences since the method's signature doesn't contain parameters and the caching mechanism doesn't look like it differentiates based on the object's hashCode so all instances of UserReference would share the same cache instance unless I'm reading the docs wrong. We'd probably need to get the cache to use the private id field to cache, but I don't see a way to do that out of the box

https://docs.spring.io/spring-framework/reference/integration/cache/annotations.html#cache-annotations-cacheable-key

Reply:
The cache would be defined at the repository layer